### PR TITLE
Made simple agents suitable for challenge submissions.

### PR DESCRIPTION
**Pull request by mathfac**
## Motivation and Context
Made simple agents suitable for challenge submissions:
- switch from running benchmark to challenge
- get rid of config for simple agents as only one parameter needed
- removed redundant act method

## How Has This Been Tested

`pytest`
`black . --line-length 79;mypy . --ignore-missing-imports;pycodestyle;flake8`

## Types of changes

- Bug fix (non-breaking change which fixes an issue)/improvement

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: baselines/agents/simple_agents.py [hunk 1]
```diff
@@ -58,9 +53,8 @@ def act(self, observations):
 
 
 class RandomForwardAgent(RandomAgent):
-    def __init__(self, config):
-        super(RandomForwardAgent, self).__init__(config)
-        self.dist_threshold_to_stop = config.TASK.SUCCESS_DISTANCE
+    def __init__(self, success_distance):
+        super(RandomForwardAgent, self).__init__(success_distance)
```
Comment (abhiskk) [inline baselines/agents/simple_agents.py]: You can do `super().__init__(success_distance)` (Python3 syntax)

File: baselines/agents/simple_agents.py [hunk 2]
```diff
@@ -81,8 +75,8 @@ def act(self, observations):
 
 
 class GoalFollower(RandomAgent):
-    def __init__(self, config):
-        super(GoalFollower, self).__init__(config)
+    def __init__(self, success_distance):
+        super(GoalFollower, self).__init__(success_distance)
```
Comment (abhiskk) [inline baselines/agents/simple_agents.py]: `super().__init__(success_distance)`

File: baselines/agents/simple_agents.py [hunk 3]
```diff
@@ -135,20 +129,15 @@ def get_agent_cls(agent_class_name):
 
 def main():
     parser = argparse.ArgumentParser()
-    parser.add_argument(
-        "--task-config", type=str, default="tasks/pointnav.yaml"
-    )
+    parser.add_argument("--success_distance", type=float, default=0.2)
```
Comment (abhiskk) [inline baselines/agents/simple_agents.py]: "--success_distance" -> `--success-distance`. Note that this does not require changes in python code, python argpase automatically maps `--success-distance` to `args.success_distance`.

File: baselines/agents/simple_agents.py [hunk 4]
```diff
@@ -135,20 +129,15 @@ def get_agent_cls(agent_class_name):
...
     agent = get_agent_cls(args.agent_class)(
-        habitat.get_config(args.task_config)
+        success_distance=args.success_distance
     )
-    benchmark = habitat.Benchmark(args.task_config)
-    metrics = benchmark.evaluate(agent)
-
-    for k, v in metrics.items():
-        habitat.logger.info("{}: {:.3f}".format(k, v))
+    challenge = habitat.Challenge()
```
Comment (abhiskk) [inline baselines/agents/simple_agents.py]: I suggest we retain the old `benchmark` usage here instead of `challenge`. This code is orthogonal to challenge codebase I think, also if someone is testing locally `benchmark` fits in better semantically compared to `challenge`

File: baselines/agents/simple_agents.py [hunk 5]
```diff
@@ -135,20 +129,15 @@ def get_agent_cls(agent_class_name):
 
 def main():
     parser = argparse.ArgumentParser()
-    parser.add_argument(
-        "--task-config", type=str, default="tasks/pointnav.yaml"
-    )
+    parser.add_argument("--success_distance", type=float, default=0.2)
     parser.add_argument("--agent_class", type=str, default="GoalFollower")
```
Comment (abhiskk) [inline baselines/agents/simple_agents.py]: Same as above `--agent-class`

File: baselines/agents/simple_agents.py (hunk 3 above)
Comment (mathfac) [reply baselines/agents/simple_agents.py]: @abhiskk can you give more context why `success-distance` is better than `success_distance`?

Comment (abhiskk) [reply baselines/agents/simple_agents.py]: It is not necessarily better, but generally I have seen `success-distance` type cmd arguments being used in python, I looked around though and I have found both types of examples actually so don't have a preference here. We should be consistent, the only other place I think command line args are being used in the project are for PPO: https://github.com/facebookresearch/habitat-api/blob/master/baselines/rl/ppo/utils.py#L275  and they are in the format of `--success-distance`. I am happy with either usage, but we should be consistent.

Comment (erikwijmans) [reply baselines/agents/simple_agents.py]: I tend to think that `--success-distance` looks cleaner, although it makes it a little harder to search for the definition of the command line arg for a certain variable.

**Comment by mathfac** [comment]
Addressed the comments.

**Comment by erikwijmans** [comment]
Looks like a `.swp` file got added with this PR :(
